cdap-api/src/main/java/co/cask/cdap/api/dataset/lib/PartitionedFileSetArguments.java

```diff
@@ -170,4 +172,15 @@ public static Comparable convertFieldValue(String where, String kind, String fie
                       where, kind, stringValue, fieldName, fieldType.name()), e);
     }
   }
+
+  public static void addPartitions(Map<String, String> arguments, Iterator<Partition> partitionIterator) {
```

Comment:
Why does this take a partition and not a partition key? Since runtime arguments come from outside, we can't assume that the caller has an instance of the dataset that he can call getPartition() on. It is more generic to accept a partition key. Didn't we discuss that, that a PartitionKey is a special case of a PartitionFilter, where all fields are specified as exact values?